crawl-ref/source/state.h

```diff
@@ -139,6 +139,16 @@ struct game_state
     // Should flushing a nonempty key buffer error or crash? Used for tests.
     bool nonempty_buffer_flush_errors;
 
+    // Whether we are in "map mode", i.e. currently in eXamine
+    bool in_map_mode;
+
+    // If we are in map mode, absolute level pos of the cursor (position and branch id)
```

Comment:
Would it be cleaner to bundle these up into a struct? Also, I'm not super happy about these fields which are always set but might be meaningless/undefined if `in_map_mode` is false. We probably can't do anything much better, lacking std::optional, but if you can see an improvement I'd love to.

Reply:
I do like the idea of having all the info about map mode be in one struct; I do think the boolean should be separate though. As for the undefinedness problem, this struct could be served through a getter, but I think that's somewhat overcomplicating things.

Reply:
structified the latter two variables, let me know what you think